### PR TITLE
DECKHAND-10: Add Barbican integration to Deckhand

### PR DESCRIPTION
**Pull request by fmontei**
This commit adds Barbican integration to Deckhand.

TODO:
  - properly integrate with keystone -- currently relying on hardcoded values in deckhand.conf.config
  - integrate with oslo.log
  - properly build out API -- deckhand.control.secrets is currently a dummy API (
    it successfully returns the list of CAs stored by Barbican)
  - update README to include Barbican information

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 falcon==1.1.0
+
+keystoneauth1>=2.21.0  # Apache-2.0
 oslo.config>=3.22.0 # Apache-2.0
-oslo.config>=3.22.0  # Apache-2.0
+oslo.context>=2.14.0 # Apache-2.0
```
Comment (mark-burnett) [inline requirements.txt:5]: This is not directly a comment for this PR, but I'd like to separate our indirect dependencies from our direct dependencies so that we can maintain them more easily.  One way we could approach this would be to maintain two requirements files, one with direct dependencies and one with indirect dependencies.  Then we can have a tox task that updates the full set (requirements.txt).  This should give us good control over the libraries we are using.

Also, I think we need to specify all our dependencies exactly via `==` rather than with ranges (again, this is not directly related to this PR).

File: README.rst [hunk 2]
```diff
@@ -9,5 +9,4 @@ To run::
 	$ virtualenv -p python3 /var/tmp/deckhand
 	$ . /var/tmp/deckhand/bin/activate
 	$ sudo pip install .
```
Comment (mark-burnett) [inline README.rst:11]: This comment is not directly related to this PR.  We should probably avoid using sudo for these commands,

**Review by mark-burnett — APPROVED** [review]
LGTM, I'm going to let someone also look + merge.

File: requirements.txt (hunk 1 above)
Comment (fmontei) [reply requirements.txt:5]: Thanks for your feedback. I'll do that in a follow-up commit.

File: README.rst [hunk 3]
```diff
@@ -9,5 +9,4 @@ To run::
 	$ virtualenv -p python3 /var/tmp/deckhand
```
Comment (sh8121att) [inline README.rst:9]: It looks like you adopted the pattern I used in Drydock. In the near future we should move all the UCP services to demonstrate usage based on containers. This offers the same isolation as virtualenvs and is closer to how they will be used in production. As mark-burnett said, not specific to this PR and can be addressed in the future.

File: requirements.txt (hunk 1 above)
Comment (sh8121att) [reply requirements.txt:5]: +1 to split dependencies

**Review by sh8121att — APPROVED** [review]
A few comments, but nothing to stand in the way of merging this.
